src/circuit.rs

```diff
@@ -183,591 +260,24 @@ impl Circuit {
-                layouter.namespace(|| "witness psi_new"),
-                config.advices[0],
-                self.psi_new,
-            )?;
-
-            let rcm_new = ScalarFixed::new(
-                ecc_chip,
-                layouter.namespace(|| "rcm_new"),
-                self.rcm_new.as_ref().map(|rcm_new| rcm_new.inner()),
-            )?;
-
-            // g★_d || pk★_d || i2lebsp_{64}(v) || i2lebsp_{255}(rho) || i2lebsp_{255}(psi)
-            let cm_new = gadget::note_commit(
-                layouter.namespace(|| {
-                    "g★_d || pk★_d || i2lebsp_{64}(v) || i2lebsp_{255}(rho) || i2lebsp_{255}(psi)"
-                }),
-                config.sinsemilla_chip_2(),
-                config.ecc_chip(),
-                config.note_commit_chip_new(),
-                g_d_new.inner(),
-                pk_d_new.inner(),
-                v_new.clone(),
-                rho_new,
-                psi_new,
-                rcm_new,
-            )?;
-
-            let cmx = cm_new.extract_p();
-
-            // Constrain cmx to equal public input
-            layouter.constrain_instance(cmx.inner().cell(), config.primary, CMX)?;
+            additional_zsa_witnesses,
         }
-
-        // Constrain the remaining Orchard circuit checks.
-        layouter.assign_region(
-            || "Orchard circuit checks",
-            |mut region| {
-                v_old.copy_advice(|| "v_old", &mut region, config.advices[0], 0)?;
-                v_new.copy_advice(|| "v_new", &mut region, config.advices[1], 0)?;
-                v_net_magnitude_sign.0.copy_advice(
-                    || "v_net magnitude",
-                    &mut region,
-                    config.advices[2],
-                    0,
-                )?;
-                v_net_magnitude_sign.1.copy_advice(
-                    || "v_net sign",
-                    &mut region,
-                    config.advices[3],
-                    0,
-                )?;
-
-                root.copy_advice(|| "calculated root", &mut region, config.advices[4], 0)?;
-                region.assign_advice_from_instance(
-                    || "pub input anchor",
-                    config.primary,
-                    ANCHOR,
-                    config.advices[5],
-                    0,
-                )?;
-
-                region.assign_advice_from_instance(
-                    || "enable spends",
-                    config.primary,
-                    ENABLE_SPEND,
-                    config.advices[6],
-                    0,
-                )?;
-
-                region.assign_advice_from_instance(
-                    || "enable outputs",
-                    config.primary,
-                    ENABLE_OUTPUT,
-                    config.advices[7],
-                    0,
-                )?;
-
-                config.q_orchard.enable(&mut region, 0)
-            },
-        )?;
-
-        Ok(())
     }
 }
 
 /// The verifying key for the Orchard Action circuit.
```

Comment:
This documentation needs to be updated reflect the fact that in the current type system, this could be a verifying key for either the original Orchard Action circuit, or the OrchardZSA circuit.

The changelog also needs a "Changed" entry saying that the semantics of `VerifyingKey` have changed in this way.

We may also want to consider whether `VerifyingKey` should instead have the generic parameter `<C: OrchardCircuit>`, but IDK right now what the wider implications of this are.